common/src/main/java/com/microsoft/identity/common/internal/fido/FidoChallengeFactory.kt

```diff
@@ -0,0 +1,112 @@
+            return AuthFidoChallenge(
+                challenge = validateRequiredParameter(
+                    FidoRequestField.Challenge.name,
+                    parameters[FidoRequestField.Challenge.name]
+                ),
+                relyingPartyIdentifier = validateRequiredParameter(
+                    FidoRequestField.RelyingPartyIdentifier.name,
+                    parameters[FidoRequestField.RelyingPartyIdentifier.name]
+                ),
+                userVerificationPolicy = validateRequiredParameter(
+                    FidoRequestField.UserVerificationPolicy.name,
+                    parameters[FidoRequestField.UserVerificationPolicy.name]
+                ),
+                version = validateRequiredParameter(
+                    FidoRequestField.Version.name,
+                    parameters[FidoRequestField.Version.name]
+                ),
+                submitUrl = validateRequiredParameter(
+                    FidoRequestField.SubmitUrl.name,
+                    parameters[FidoRequestField.SubmitUrl.name]
+                ),
+                keyTypes = validateRequiredListParameter(
+                    FidoRequestField.KeyTypes.name,
+                    parameters[FidoRequestField.KeyTypes.name]
+                ),
+                context = validateRequiredParameter(
+                    FidoRequestField.Context.name,
+                    parameters[FidoRequestField.Context.name]
+                ),
+                allowedCredentials = validateRequiredListParameter(
+                    AuthFidoRequestField.AllowedCredentials.name,
+                    parameters[AuthFidoRequestField.AllowedCredentials.name]
+                )
+            )
+        }
+
+        /**
+         * Validates that the given parameter is not null or empty.
+         * @param field passkey protocol field
+         * @param value value for a passkey protocol parameter
+         * @throws ClientException if the parameter is null or empty.
+         */
+        @Throws(ClientException::class)
+        private fun validateRequiredParameter(field: String, value: String?): String {
+            if (value == null) {
+                throw ClientException(PASSKEY_PROTOCOL_REQUEST_INVALID, "$field not provided")
+            } else if (value.isBlank()) {
+                throw ClientException(PASSKEY_PROTOCOL_REQUEST_INVALID, "$field is empty")
+            }
+            return value;
+        }
+
+        /**
+         * Validates that the given parameter is not null or empty.
+         * @param value list value for a passkey protocol parameter
+         * @param field passkey protocol field
+         * @throws ClientException if the parameter is null or empty
+         */
+        @Throws(ClientException::class)
+        private fun validateRequiredListParameter(field: String, value: String?): List<String> {
+            return validateRequiredParameter(field, value).split(DELIMITER).toList()
```

Comment:
Not sure what we are doing here. Do we want to check each individual element of the list? Seems like we are only checking it as a full string then splitting it

Reply:
It was only meant to check as full string and split, but checking individual elements against known values when possible is something I'll do in a future PR as well (becaues I need the complete server side design to know what to check against)